lnst/Common/Parameters.py

```diff
@@ -179,6 +180,53 @@ def type_check(self, value):
         return value
 
 
+class NetworkParam(Param):
+    """Network Param
+
+    This parameter is used to specify the IP networks a test will have
+     interfaces on. Using it allows multiple simultaneous LNST sessions on a shared broadcast domain.
+
+     It is used with physical or created logical interfaces (vlans, bonds, etc) have
+     addresses on external networks.
+
+     Things like tunnel recipes generally use it for interfaces in the underlying network.
+
+     For example here is VlanRecipe use of it::
+        vlan0_id = IntParam(default=10)
+        vlan1_id = IntParam(default=20)
+        vlan2_id = IntParam(default=30)
+
+        vlan0_ipv4 = IPv4NetworkParam(default="192.168.10.0/24")
+        vlan0_ipv6 = IPv6NetworkParam(default="fc00:0:0:1::/64")
+
+        vlan1_ipv4 = IPv4NetworkParam(default="192.168.20.0/24")
+        vlan1_ipv6 = IPv6NetworkParam(default="fc00:0:0:2::/64")
+
+        vlan2_ipv4 = IPv4NetworkParam(default="192.168.30.0/24")
+        vlan2_ipv6 = IPv6NetworkParam(default="fc00:0:0:3::/64")
```

Comment:
Maybe adding an example together with the `interface_address` generator would be more helpful. Also, to be less verbose here, a SimpleNetworkRecipe example would be enough I think.

Reply:
I kinda wanted to put a slightly less trivial example that demonstrates multiple networks. I always hate it when docs only give simplistic examples. Although now that I've gone through the recipes I found a couple 2 network ones that might be better.

Reply:
Agreed on `interface_addresses` I could probably give a better example there.

Reply:
> I always hate it when docs only give simplistic examples.

Sure, but you should start with a simple explanation and then build on top of that for more complex use.

The problem I see with the current documentation is that it's simply missing basic explanation what the parameter type holds. You jumped right into specific use cases, which imo is less important.

So, I'd start with following.

`NetwokParam` is a parameter type that can hold either IPv4 or IPv6 network address, specified as string. It automatically checks correctness of the passed network address.

When used together with `interface_address` it provides a convenient way to generate network addresses for the host interfaces used in LNST recipes.

For example, ....

... then you can provide more specific examples. But imo, that information is sufficient.

Reply:
Fixed the interface_addresses and NetworkParams doc string let me know what you think